### PR TITLE
Set custom Authorization Token through generateAuthHeader class method

### PR DESCRIPTION
**Pull request by fmquaglia**
[Implemented](https://github.com/fmquaglia/jsorm/blob/configure_token_format/src/model.ts#L170-L172) the `generateAuthHeader` class method in `Model` which can be overridden in the desired class. 

(Sorry if I'm not using proper TypeScript terminology, I actually don't know much about it).

### CODE REVIEW

File: src/model.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ export default class Model {
     }
 
     if (this.getJWT()) {
-      options.headers.Authorization = `Token token="${this.getJWT()}"`;
+        options.headers.Authorization = this.generateAuthHeader(this.getJWT());
```
Comment (richmolj) [inline src/model.ts]: 2 spaces not 4 please :)

File: test/fixtures.ts [hunk 2]
```diff
@@ -97,6 +97,7 @@ class MultiWord extends ApplicationRecord {
 
 const TestJWTSubclass = ApplicationRecord.extend({});
 
+
```
Comment (richmolj) [inline test/fixtures.ts]: excess line

File: test/integration/authorization-test.ts [hunk 3]
```diff
@@ -27,6 +27,30 @@ describe('authorization headers', function() {
...
+        return `Bearer ${token}`;
+      };
+    });
+
+    afterEach(function() {
+      fetchMock.restore();
+      Person.generateAuthHeader = originalHeaderFn;
+    });
+
+    it('expects true to be true', function(done) {
```
Comment (richmolj) [inline test/integration/authorization-test.ts]: let's make this a meaningful test description please

Comment (fmquaglia) [reply test/integration/authorization-test.ts]: 🤦‍♂️ @richmolj sorry about that. I was struggling with `fetchMock` and started by clearing any chance to have an error on the setup. Thank you. Sorry. And thank you again.

**Comment by fmquaglia** [comment]
Hey, I made the necessary adjustments based on your remarks @richmolj 
Let me know if you need anything else from me.

File: test/integration/authorization-test.ts [hunk 4]
```diff
@@ -27,6 +27,30 @@ describe('authorization headers', function() {
...
+        return `Bearer ${token}`;
+      };
+    });
+
+    afterEach(function() {
+      fetchMock.restore();
+      Person.generateAuthHeader = originalHeaderFn;
+    });
+
+    it('sends the custom Authorization token in the request¡\'s headers', function(done) {
```
Comment (richmolj) [inline test/integration/authorization-test.ts]: So tiny, but could you make the outer quotes `"` so the inner quote can be `'` without escaping? After that we are good to go!

**Comment by richmolj** [comment]
Published in 0.5.3, thanks @fmquaglia !

**Comment by fmquaglia** [comment]
My pleasure @richmolj 
Thank YOU!

**Pull request merged**